receiver/awscontainerinsightreceiver/internal/k8sapiserver/k8sapiserver.go

```diff
@@ -232,18 +244,28 @@ func (k *K8sAPIServer) init() error {
 		k.logger.Info(fmt.Sprintf("configMap: %v, err: %v", configMap, err))
 	}
 
-	lock, err := resourcelock.New(
-		resourcelock.ConfigMapsLeasesResourceLock,
-		lockNamespace, k.leaderLockName,
-		clientSet.CoreV1(),
-		clientSet.CoordinationV1(),
-		resourcelock.ResourceLockConfig{
-			Identity:      k.nodeName,
-			EventRecorder: k.createRecorder(k.leaderLockName, lockNamespace),
-		})
-	if err != nil {
-		k.logger.Warn("Failed to create resource lock", zap.Error(err))
-		return err
+	var lock resourcelock.Interface
+	if k.leaderLockUsingConfigMapOnly {
+		lock = &ConfigMapLock{
+			ConfigMapMeta: metav1.ObjectMeta{
+				Namespace: lockNamespace,
+				Name:      k.leaderLockName,
+			},
+			Client:     clientSet.CoreV1(),
+			LockConfig: resourceLockConfig,
+		}
+	} else {
+		l, err := resourcelock.New(
```

Comment:
I see why we would change it to not set `lock` to the new value if there is an error, but that is a behavior change from what was there before. Do you see any concerns with that?

Reply:
I hadnt actually given it much thought mainly because if there is an err, it would just return and not use lock anymore at that point.

But to keep that behavior the same as before, I guess I can just move `lock = l` to be before the err check on line 264? Shouldnt matter if we set it since it would just return on line 266 if there was an err.

Reply:
That's my bad. I had to expand the code. I thought that lock was defined somewhere as a global, and this was changing how it got set. But it's local to this function. Disregard.